schema.ts

```diff
@@ -1201,7 +1211,24 @@ export interface ObjectSchemaDefinition<K extends string, L extends string>
    * @hidden
    */
   modifiedByProperty?: PropertyIdentifier<K>;
-
+  /**
+   * For cases where the object being synced represents a user, the name of the property within
+   * {@link ObjectSchemaDefinition.properties} that identifies the email address of the user.
+   *
+   * Must be a {@link ValueType.String} property with the {@link ValueHintType.Email} hint or
+   * a {@link ValueType.Object} with the {@link ValueHintType.Person} hint
+   * @hidden
+   */
+  userEmailProperty?: PropertyIdentifier<K>;
```

Comment:
I wonder if we could make this more generic, like: 
- `emailProperty`
- `primaryEmailProperty`
- `contactEmailProperty`
Then we could use the same field for tables with other roles (`Group`, etc).

Reply:
Discussed f2f. Will likely leave as is here for now